### ISSUE TITLE
Error during upgrade, moving courses from table  turnitintooltwo_courses to table plagiarism_turnitin_courses

### ISSUE THREAD

**Issue by steveorulez** [turnitin/moodle-plagiarism_turnitin#462]
Hello,
during the upgrade to the new version of the plugin i get this error:

Did you remember to make the first column something unique in your call to get_records? Duplicate value '3050' found in column 'courseid'.

    line 1254 of /lib/dml/mysqli_native_moodle_database.php: call to debugging()
    line 1400 of /lib/dml/moodle_database.php: call to mysqli_native_moodle_database->get_records_sql()
    line 1351 of /lib/dml/moodle_database.php: call to moodle_database->get_records_select()
    line 337 of /plagiarism/turnitin/db/upgrade.php: call to moodle_database->get_records()
    line 636 of /lib/upgradelib.php: call to xmldb_plagiarism_turnitin_upgrade()
    line 1852 of /lib/upgradelib.php: call to upgrade_plugins()
    line 694 of /admin/index.php: call to upgrade_noncore()

Did you remember to make the first column something unique in your call to get_records? Duplicate value '7178' found in column 'courseid'.

    line 1254 of /lib/dml/mysqli_native_moodle_database.php: call to debugging()
    line 1400 of /lib/dml/moodle_database.php: call to mysqli_native_moodle_database->get_records_sql()
    line 1351 of /lib/dml/moodle_database.php: call to moodle_database->get_records_select()
    line 337 of /plagiarism/turnitin/db/upgrade.php: call to moodle_database->get_records()
    line 636 of /lib/upgradelib.php: call to xmldb_plagiarism_turnitin_upgrade()
    line 1852 of /lib/upgradelib.php: call to upgrade_plugins()
    line 694 of /admin/index.php: call to upgrade_noncore()

The problem is happening because on 2 courses i have more then 1 assigment with plagiarism plugin enabled from different ownerid:
$ppcourses = $DB->get_records('turnitintooltwo_courses', array('course_type' => 'PP'), 'id ASC', 'courseid, ownerid, turnitin_ctl, turnitin_cid');

the easy fix should be to add the column id in the list of fields but there is a unique index on the courseid column. What does it happen if i manually delete the extra records from the table before starting the upgrade?

**Comment by carl-hostrander** [comment]
Thank you for reporting this.  Because the latest version of the plagiarism plugin is supported for versions of Moodle 4.1 and higher, I am closing this ticket.  However, if you find this issue is occurring with the latest version of the plugin in any of the supported Moodle versions, please create a new ticket and we will address it.
